internal/versions/versions.go

```diff
@@ -92,37 +121,238 @@ func (h *handler) GetMustGatherImages(openshiftVersion, cpuArchitecture, pullSec
+	if err != nil {
+		return nil, err
+	}
+
+	supportLevels, err := common.GetOpenshiftVersionsSupportLevelFromDBWhere(h.db, "openshift_version = ?", majorMinorVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(supportLevels) == 0 {
+		return nil, errors.Errorf("error occurred while trying to get the support level of %s from DB", *release.Version)
+	}
+
+	return swag.String(supportLevels[0].SupportLevel), nil
+}
+
+func (h *handler) createReleaseImageFromDBOpenshiftVersion(
+	dbVersion *common.OpenshiftVersion,
+	cpuArch string,
+) (*models.ReleaseImage, error) {
+	if dbVersion == nil {
+		return nil, nil
+	}
+
+	releaseModel := models.ReleaseImage{
+		CPUArchitecture:  &cpuArch,
+		CPUArchitectures: []string{cpuArch},
+		Default:          dbVersion.Default,
+		OpenshiftVersion: dbVersion.DisplayName,
+		Version:          dbVersion.DisplayName,
+		SupportLevel:     "",
+		URL:              h.getURLForDBRelease(*dbVersion, cpuArch),
+	}
+
+	if cpuArch == common.MultiCPUArchitecture {
+		releaseModel.CPUArchitectures = supportedMultiArchitectures
+	}
+
+	if dbVersion.SupportLevel != nil {
+		releaseModel.SupportLevel = *dbVersion.SupportLevel
+		return &releaseModel, nil
+	}
+
+	newSuppotLevel, err := h.getReleaseSupportLevel(releaseModel)
+	if err != nil {
+		h.log.WithError(err).Debug("error occurred while trying to get openshift releases support levels")
+		return nil, err
+	}
+	releaseModel.SupportLevel = *newSuppotLevel
+
+	return &releaseModel, nil
+}
+
+func (h *handler) getReleaseImageFromDB(
+	openshiftVersion,
+	cpuArchitecture string,
+	versionSegmentsLength int,
+) (*models.ReleaseImage, error) {
+	openshiftVersion = strings.TrimSuffix(openshiftVersion, "-multi")
+
+	var (
+		dbOpenshiftVersions []*common.OpenshiftVersion
+		targetVersion       *common.OpenshiftVersion
+		err                 error
+	)
+
+	// specific version
+	if versionSegmentsLength >= 3 {
+		targetVersion, err = common.GetOpenshiftVersionFromDBWhere(
+			h.db,
+			"display_name = ? AND ? = ANY(cpu_architectures)",
+			openshiftVersion,
+			cpuArchitecture,
+		)
+		if err != nil {
+			h.log.WithError(err).
+				Debugf(
+					"error occurred while trying to get openshift release image for version %s and CPU architecture %s from the DB",
+					openshiftVersion,
+					cpuArchitecture,
+				)
+			return nil, err
+		}
+
+		return h.createReleaseImageFromDBOpenshiftVersion(targetVersion, cpuArchitecture)
+	}
+
+	// get latest release for major.minor or major
+	dbOpenshiftVersions, err = h.getReleasesByPartVersionAndArchitectureFromDB(openshiftVersion, cpuArchitecture)
+	if err != nil {
+		return nil, err
+	}
+
+	targetVersion, err = h.getLatestDBRelease(dbOpenshiftVersions)
+	if err != nil {
+		return nil, err
+	}
+
+	return h.createReleaseImageFromDBOpenshiftVersion(targetVersion, cpuArchitecture)
+}
+
+func (h *handler) getLatestDBRelease(releases []*common.OpenshiftVersion) (*common.OpenshiftVersion, error) {
```

Comment:
I don't think we should have two different types.

Ideally we would use ReleaseImage for the DB.